### PR TITLE
Alerts v2

### PR DESCRIPTION
**Pull request by mirii1994**
This PR:
* Adds new resource Alerts V2.
* Adds new datasource Alerts V2.
* Adds a 4 sec delay to the existing alert to avoid 404 error.
* Testing, examples, docs for the new resource & datasource.
* Updates provider to use new client version.
* Deletes upgrade plugin script.

### CODE REVIEW

File: logzio/validators.go [hunk 1]
```diff
@@ -80,3 +111,32 @@ func validateSeverityTypes(v interface{}, k string) (ws []string, errors []error
...
+		alerts_v2.SortAsc,
+		alerts_v2.SortDesc,
+	}
+
+	if !contains(validTypes, value){
+		errors = append(errors, fmt.Errorf("severity type %q must be one of %s", k, validTypes))
+	}
+
+	return
+}
```
Comment (yyyogev) [inline logzio/validators.go:142]: Beautiful!

**Review by yyyogev — APPROVED** [review]
Cheers! 🥂

File: logzio/resource_alert_v2.go [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+		}
+
+		return err
+	}
+
+	alertId := strconv.FormatInt(a.AlertId, BASE_10)
+	d.SetId(alertId)
+
+	// A temporary solution to handle a 404 error that happens otherwise
+	time.Sleep(delayGet * time.Second)
```
[the remark this answers is not in the document]
Comment (mirii1994) [reply logzio/resource_alert_v2.go]: Used the retry-go module in the new alerts resource

File: logzio/resource_alert_v2.go [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+			alertV2FilterMustNot: subComponent.QueryDefinition.Filters.Bool.MustNot,
+			alertV2GroupBy: subComponent.QueryDefinition.GroupBy,
+			alertV2AggregationField: subComponent.QueryDefinition.Aggregation.FieldToAggregateOn,
+			alertV2AggregationType: subComponent.QueryDefinition.Aggregation.AggregationType,
+			alertV2ShouldQueryOnAllAccounts: subComponent.QueryDefinition.ShouldQueryOnAllAccounts,
+			alertV2AccountIdsToQuery: subComponent.QueryDefinition.AccountIdsToQueryOn,
+			alertV2Operation: subComponent.Trigger.Operator,
+			alertV2SeverityThresholdTiers: severityThreshold,
+			alertV2Columns: columns,
+			alertV2OutputShouldUseAllFields: subComponent.Output.ShouldUseAllFields,
```
[the remark this answers is not in the document]
Comment (mirii1994) [reply logzio/resource_alert_v2.go:414]: Had to do some manipulations to the schema so unmarshal won't work here

File: logzio/resource_alert_v2.go [hunk 4]
```diff
@@ -292,9 +294,19 @@ func resourceAlertV2Create(d *schema.ResourceData, m interface{}) error {
...
-	// A temporary solution to handle a 404 error that happens otherwise
-	time.Sleep(delayGet * time.Second)
-	return resourceAlertV2Read(d, m)
+	return retry.Do(
+		func() error {
+			return resourceAlertV2Read(d, m)
+		},
+		retry.RetryIf(
+			func(err error) bool {
+				if strings.Contains(err.Error(), "missing alert"){
```
Comment (yyyogev) [inline logzio/resource_alert_v2.go]: does the err msg contains "404"? if so, better to search for it and not "missing alert"

File: logzio/resource_alert_v2.go [hunk 5]
```diff
@@ -292,9 +294,19 @@ func resourceAlertV2Create(d *schema.ResourceData, m interface{}) error {
...
+					return true
+				}
+				return false
+			}),
+		retry.Delay(delayGetAlertV2),
```
Comment (yyyogev) [inline logzio/resource_alert_v2.go:308]: If you want, you could do delayFunc so the delays won't be a second each but 1,2,4.. or 1,1,2,3,5.. 
Anyway, please set max attempts for the retries

File: logzio/resource_alert_v2.go [hunk 6]
```diff
@@ -341,9 +353,20 @@ func resourceAlertV2Update(d *schema.ResourceData, m interface{}) error {
...
-	time.Sleep(delayGet * time.Second)
-	return resourceAlertV2Read(d, m)
+	return retry.Do(
+		func() error {
+			return resourceAlertV2Read(d, m)
+		},
+		retry.RetryIf(
+			func(err error) bool {
+				createAlert := createCreateAlertType(d)
+				if !reflect.DeepEqual(createAlert, updateAlert) || err != nil {
```
Comment (yyyogev) [inline logzio/resource_alert_v2.go:363]: Cool!

**Review by yyyogev — APPROVED** [review]
It's really up to you now but I personally think a 500+ lines class is not ideal 🤷

File: logzio/resource_alert_v2.go (hunk 4 above)
Comment (mirii1994) [reply logzio/resource_alert_v2.go]: The client returns the following message: "API call GetAlert failed with missing alert %d, data: %s"

**Pull request merged**